client/src/components/Frontend/TextFilter.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { inject } from 'lib/Injector';
+import classnames from 'classnames';
+
+/**
+ * A text filter is used to filter a CKAN dataset by a specified text string
+ * against the columns that
+ */
+const TextFilter = (props) => {
+  const {
+    columns,
+    extraClass,
+    id,
+    label,
+    TextFieldComponent,
+  } = props;
+
+  const classes = classnames(extraClass, 'form-group', 'ckan-registry__text-filter');
+  const inputId = `TextFilter_${id}_Search`;
+
+  return (
+    <div className={classes}>
+      { label && <label htmlFor={inputId}>{label}</label> }
+      <TextFieldComponent
+        id={inputId}
+        name={`TextFilter[${id}][Search]`}
+      />
+
+      <input
+        type="hidden"
+        name={`TextFilter[${id}][Columns]`}
+        value={JSON.stringify(columns)}
+      />
+    </div>
+  );
+};
+
+TextFilter.propTypes = {
+  id: PropTypes.string.isRequired,
+  columns: PropTypes.arrayOf(PropTypes.string),
+  extraClass: PropTypes.string,
+  label: PropTypes.string,
+};
+
+TextFilter.defaultProps = {
+  columns: [],
+  extraClass: '',
+  label: '',
+};
+
+export { TextFilter as Component };
+
+export default inject(
```

Comment:
Do we want to use Injector on the front end components? Considering our advice will be "just copy the code and edit for customisations" I feel that keeping it simple is best.

Reply:
Good point - I didn't even think about it. I guess we should just render inputs directly instead

Reply:
I guess there's nothing to prevent us from having Reactstrap as a dependency though.